src/main/groovy/io/github/groovymc/modsdotgroovy/MDGExtension.groovy

```diff
@@ -25,49 +25,70 @@
 package io.github.groovymc.modsdotgroovy
 
 import groovy.transform.CompileStatic
+import org.gradle.api.Project
+import org.gradle.api.provider.ListProperty
 import org.gradle.api.provider.Property
+import org.gradle.api.tasks.SourceSet
 
 @CompileStatic
 abstract class MDGExtension {
     public static final String NAME = 'modsDotGroovy'
     abstract Property<String> getDslVersion()
     abstract Property<Boolean> getAutomaticConfiguration()
-    abstract Property<Type> getType()
+    abstract ListProperty<Platform> getPlatforms()
+    abstract Property<SourceSet> getSource()
 
-    MDGExtension() {
+    protected final Project project
+
+    MDGExtension(final Project project) {
+        this.project = project
         automaticConfiguration.set(true)
-        type.set(Type.FORGE)
+        platforms.set([])
     }
 
     String mdgDsl(String version = null) {
         version = version ?: getDslVersion().get()
         return "io.github.groovymc.modsdotgroovy:dsl:$version"
     }
 
-    void type(String type) {
-        switch (type.toLowerCase(Locale.ROOT)) {
-            case 'quilt':
-                this.type.set(Type.QUILT)
-                break
-            case 'forge':
-                this.type.set(Type.FORGE)
-                break
-            default:
-                throw new IllegalArgumentException("Unknown project type :$type")
-        }
+    void platform(String name) {
+        this.platforms.set([Platform.byName(name)])
+    }
+
+    void platform(List<String> platforms) {
```

Comment:
```suggestion
    void platforms(List<String> platforms) {
```

Reply:
Here's the issue I've been having here. For some reason, the following code (without that change, or with it and using `platforms`) in a build.gradle doesn't work:
```gradle
platform ['quilt', 'forge']
```
It tries to do something by finding the property names `platform` instead of running the method

Reply:
Oh wait, it's trying to do `getAt` on the property. Ah well